### ISSUE TITLE
Problem with callbacks in different tabs

### ISSUE THREAD

**Issue by delaleva** [plotly/dash-core-components#283]
I’ve got two tabs in my applications and each one of them has a graph, a dropdown and a callback that updates the graph. The problem is that when I go to the second tab and try to update the graph, I’m automatically taken to the first tab which is the default and then I have to click on the second tab again to see the changes. I tried using Div as suggested by someone in the Plotly forum but it didn't work. I suspect that's a bug.

My original code:

```
import dash
import dash_html_components as html
import dash_core_components as dcc
import pandas as pd
from dash.dependencies import Input, Output
import plotly.graph_objs as go


app = dash.Dash()

df = pd.read_csv('gapminderDataFiveYear.csv')


year_options = []
for year in df['year'].unique():
    year_options.append({'label':str(year),'value':year})

# Create a Dash layout
app.layout = html.Div([
    html.Div(
        html.H1('My Dashboard')
    ),
    dcc.Tabs(id="tabs", value='Tab1', children=[
        dcc.Tab(label='Tab 1', id='tab1', value='Tab1', children =[
            dcc.Graph(id='graph-1'),
            dcc.Dropdown(id='year-picker-1', options=year_options, value=df['year'].min())


        ]),

        dcc.Tab(label='Tab 2', id='tab2', value= 'Tab2', children=[
            dcc.Graph(id='graph-2'),
            dcc.Dropdown(id='year-picker-2', options=year_options, value=df['year'].min())


        ])
    ])
])


@app.callback(Output('graph-1', 'figure'),
              [Input('year-picker-1', 'value')])
def update_figure_1(selected_year):
    filtered_df = df[df['year'] == selected_year]
    traces = []
    for continent_name in filtered_df['continent'].unique():
        df_by_continent = filtered_df[filtered_df['continent'] == continent_name]
        traces.append(go.Scatter(
            x=df_by_continent['gdpPercap'],
            y=df_by_continent['lifeExp'],
            text=df_by_continent['country'],
            mode='markers',
            opacity=0.7,
            marker={'size': 15},
            name=continent_name
        ))

    return {
        'data': traces,
        'layout': go.Layout(
            xaxis={'type': 'log', 'title': 'GDP Per Capita'},
            yaxis={'title': 'Life Expectancy'},
            hovermode='closest'
        )
    }


@app.callback(Output('graph-2', 'figure'),
              [Input('year-picker-2', 'value')])
def update_figure_1(selected_year):
    filtered_df = df[df['year'] == selected_year]
    traces = []
    for continent_name in filtered_df['continent'].unique():
        df_by_continent = filtered_df[filtered_df['continent'] == continent_name]
        traces.append(go.Scatter(
            x=df_by_continent['gdpPercap'],
            y=df_by_continent['lifeExp'],
            text=df_by_continent['country'],
            mode='markers',
            opacity=0.7,
            marker={'size': 15},
            name=continent_name
        ))

    return {
        'data': traces,
        'layout': go.Layout(
            xaxis={'type': 'log', 'title': 'GDP Per Capita'},
            yaxis={'title': 'Life Expectancy'},
            hovermode='closest'
        )
    }


if __name__ == '__main__':
    app.run_server(debug=True)
```

Modified code using Div:

```
import dash
import dash_html_components as html
import dash_core_components as dcc
import pandas as pd
from dash.dependencies import Input, Output
import plotly.graph_objs as go


app = dash.Dash()

df = pd.read_csv('gapminderDataFiveYear.csv')


year_options = []
for year in df['year'].unique():
    year_options.append({'label':str(year),'value':year})

# Create a Dash layout
app.layout = html.Div([
    html.Div(
        html.H1('My Dashboard')
    ),
    dcc.Tabs(id="tabs", value='Tab2', children=[
        dcc.Tab(label='Tab 1', id='tab1', value='Tab1', children =[
            html.Div(id='graph-1'),
            dcc.Dropdown(id='year-picker-1', options=year_options, value=df['year'].min())


        ]),

        dcc.Tab(label='Tab 2', id='tab2', value= 'Tab2', children=[
            html.Div(id='graph-2'),
            dcc.Dropdown(id='year-picker-2', options=year_options, value=df['year'].min())


        ])
    ])
])


@app.callback(Output('graph-1', 'children'),
              [Input('year-picker-1', 'value')])
def update_figure_1(selected_year):
    graphs = []
    filtered_df = df[df['year'] == selected_year]
    traces = []
    for continent_name in filtered_df['continent'].unique():
        df_by_continent = filtered_df[filtered_df['continent'] == continent_name]
        traces.append(go.Scatter(
            x=df_by_continent['gdpPercap'],
            y=df_by_continent['lifeExp'],
            text=df_by_continent['country'],
            mode='markers',
            opacity=0.7,
            marker={'size': 15},
            name=continent_name
        ))
    graphs.append(html.Div(dcc.Graph(id = 'mygr1',figure={
        'data': traces,
        'layout': go.Layout(
            xaxis={'type': 'log', 'title': 'GDP Per Capita'},
            yaxis={'title': 'Life Expectancy'},
            hovermode='closest'
        )
    })))
    return graphs



# @app.callback(Output('graph-2', 'children'),
#               [Input('year-picker-2', 'value')])
# def update_figure_2(selected_year):
#     graphs=[]
#     filtered_df = df[df['year'] == selected_year]
#     traces = []
#     for continent_name in filtered_df['continent'].unique():
#         df_by_continent = filtered_df[filtered_df['continent'] == continent_name]
#         traces.append(go.Scatter(
#             x=df_by_continent['gdpPercap'],
#             y=df_by_continent['lifeExp'],
#             text=df_by_continent['country'],
#             mode='markers',
#             opacity=0.7,
#             marker={'size': 15},
#             name=continent_name
#         ))
#
#         graphs.append(html.Div(dcc.Graph(id='mygr2', figure={
#         'data': traces,
#         'layout': go.Layout(
#             xaxis={'type': 'log', 'title': 'GDP Per Capita'},
#             yaxis={'title': 'Life Expectancy'},
#             hovermode='closest'
#         )
#     })))
#
#     return graphs


if __name__ == '__main__':
    app.run_server(debug=True)
```

**Comment by chriddyp** [comment]
Thanks for reporting! Could you verify which version of the library you are using?
```
print(dash_core_components.__version__)
```

**Comment by delaleva** [comment]
I'm using version 0.28.0.

**Comment by valentijnnieman** [comment]
@delaleva Sorry for the delay. Could you check if `0.28.2` fixes your problems? Thanks!

**Comment by delaleva** [comment]
Yes, it does, thank you. It was fixed in version 0.28.1 as well.
